modin/pandas/dataframe.py

```diff
@@ -2155,9 +2155,19 @@ def sum(
                 dtype=pandas.api.types.pandas_dtype("object"),
             )
 
-        data = self._validate_dtypes_sum_prod_mean(
-            axis, numeric_only, ignore_axis=False
-        )
+        # We cannot add datetime types, so if we are summing a column with
+        # dtype datetime64 and cannot ignore non-numeric types, we must throw a
+        # TypeError.
+        if numeric_only is False and any(
+            dtype == pandas.api.types.pandas_dtype("datetime64[ns]")
+            for dtype in self.dtypes
+        ):
+            raise TypeError(
+                "'DatetimeArray' with dtype datetime64[ns] does not support reduction 'sum'"
+            )
+
+        data = self._get_numeric_data(axis) if numeric_only else self
```

Comment:
This piece of code has been moved from `_validate_dtypes_sum_prod_mean` function because the processing is different for sum function.